api/pull.go

```diff
@@ -207,8 +210,12 @@ func (f *findNodesInDNS) getMesosAgents() (nodes []Node, err error) {
 	}
 
 	for _, agent := range sr.Agents {
+		role := AgentRole
+		if agent.Attributes.PublicIP == "true" {
```

Comment:
Perhaps we should add a comment here to call out explicitly that we're depending on the agent's `public_ip` attribute to indicate that its resources are reserved for the public role?